op-node/rollup/derive/pipeline.go

```diff
@@ -106,6 +106,12 @@ func NewDerivationPipeline(log log.Logger, cfg *rollup.Config, l1Fetcher L1Fetch
 	}
 }
 
+// EngineReady returns true if the engine is ready to be used.
+// When it's being reset its state is inconsistent, and should not be used externally.
+func (dp *DerivationPipeline) EngineReady() bool {
+	return dp.resetting > 0
```

Comment:
`return dp.resetting >= len(dp.stages)` ?

Reply:
Only the engine stage status matters here, at position 0, the other stages are free to be in reset-state while the engine is used.